crates/js-sys/src/lib.rs

```diff
@@ -1174,6 +1174,20 @@ extern {
     pub fn next(this: &Iterator) -> Result<IteratorNext, JsValue>;
 }
 
+impl std::iter::Iterator for Iterator {
+    type Item = JsValue;
+
+    fn next(&mut self) -> Option<JsValue> {
+        let val = Iterator::next(&*self)
+            .expect("JS object didn't obey iterator protocol");
+        if val.done() {
+            None
```

Comment:
I *thought* this was wrong, and that if both done and value are present (happens when returning from a generator), that we should add the value to the iterator, but it seems that it doesn't get into the iterable in JS:

![done-and-value](https://user-images.githubusercontent.com/74571/45232264-a85e1c80-b283-11e8-8383-b5c9b8c3f56a.png)

Carry on!